File: backend/Domain/Entities/BookCopy.cs
using Domain.Common;
using Domain.Enums;

namespace Domain.Entities;

public class BookCopy : BaseEntity
{
    public string CopyCode { get; set; } = string.Empty;
    public BookCopyStatus Status { get; set; } = BookCopyStatus.Available;

    public int BookId { get; set; }
    public Book? Book { get; set; }

    public ICollection<BorrowRecord> BorrowRecords { get; set; } = new List<BorrowRecord>();
}

File: mobile/src/types/bookCopy.ts
// Khớp với Application/DTOs/Book/BookCopy/BookCopyDto.cs
export type BookCopyStatus = "Available" | "Borrowed" | "Lost" | "Damaged";

export interface BookCopy {
  id: number;
  copyCode: string;
  status: BookCopyStatus;
  bookId: number;
  bookTitle: string;
}

export interface CreateBookCopyRequest {
  copyCode: string;
  bookId: number;
}

export interface UpdateBookCopyRequest {
  copyCode: string;
  status: BookCopyStatus;
}
